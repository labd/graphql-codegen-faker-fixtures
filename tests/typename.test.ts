import { plugin } from "../dist";
import { buildSchema } from "graphql/utilities";
import { parse } from "graphql";

const schema = /* GraphQL */ `
  type Person {
    name: String
    age: String
    address: String
  }
`;

const document = /* GraphQL */ `
  fragment person on Person {
    __typename
    name
    age
  }
`;

describe("typename", () => {
  it("should create a basic builder which includes a hardcoded __typename", () => {
    const response = plugin(
      buildSchema(schema),
      [{ document: parse(document) }],
      { buildersOnly: true },
    );
    expect(response).toMatchSnapshot();
    expect(response).toContain("__typename: 'Person'");
  });
});
