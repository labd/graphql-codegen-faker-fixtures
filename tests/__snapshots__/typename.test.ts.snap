// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typename it should create a basic builder which includes a hardcoded __typename 1`] = `
"export const fakePerson = (
  override?: DeepPartial<types.PersonFragment>
): types.PersonFragment => {
  const fixture: types.PersonFragment = {
    __typename: "Person",
    name: faker.commerce.productAdjective() + " " + faker.commerce.product(),
    age: faker.lorem.words(),
  };

  return override ? merge(fixture, override) : fixture;
};
"
`;
